lib/internal/streams/destroy.js

```diff
@@ -56,7 +59,7 @@ function destroy(err, cb) {
   // If still constructing then defer calling _destroy.
   if (!s.constructed) {
     this.once(kDestroy, function(er) {
-      _destroy(this, err || er, cb);
+      _destroy(this, chainErrors(er, err), cb);
```

Comment:
please use AggregateError directly.

Reply:
I disagree, I think the current implementation is more user friendly: if there's only one error, we should return it to the user without wrapping it into an `AggregateError`.

Reply:
I mean to not use chainErrors.

Reply:
Do you mean you prefer to inline it? I'd rather not, but if you think it's the way to go, let's do that.
```suggestion
      _destroy(this, er && err ? new AggregateError([er, err]) : (err || er), cb);
```

Reply:
Also, do you mean for this specific change or all the changes in this PR?